src/main/java/inventorysetups/InventorySetupsPlugin.java

```diff
@@ -1036,10 +1029,29 @@ private boolean isNewPositionInvalid(int oldPosition, int newPosition, int size)
 
 	public List<InventorySetup> filterSetups(String textToFilter)
 	{
-		final String textToFilterLower = textToFilter.toLowerCase();
-		return inventorySetups.stream()
-			.filter(i -> i.getName().toLowerCase().contains(textToFilterLower))
-			.collect(Collectors.toList());
+    return inventorySetups.stream().filter(i -> setupContains(i, textToFilter.toLowerCase())).collect(Collectors.toList());
+	}
+
+	private static boolean setupContains(InventorySetup i, String textToFilterLower)
+	{
+		if (textToFilterLower.startsWith("item:") && textToFilterLower.length() >= 6)
+		{
+			String itemName = textToFilterLower.substring("item:".length());
+			// Find setups containing the given item name
+			return containsItem(i.getInventory(), itemName) || containsItem(i.getEquipment(), itemName)
+					|| containsItem(i.getBoltPouch(), itemName) || containsItem(i.getRune_pouch(), itemName);
+		}
+		// Find setups containing the given setup name (default behaviour)
+		return i.getName().toLowerCase().contains(textToFilterLower);
+	}
+
+	private static boolean containsItem(List<InventorySetupsItem> inventorySetupItems, String textToFilterLower) {
```

Comment:
instead of `containsItem`, maybe `containerContainsItemByName`. I would want to use `containerContainsItem` but there's a similar function for ID comparison